packages/broker/src/plugins/operator/MaintainOperatorValueHelper.ts

```diff
@@ -1,33 +1,124 @@
 import { BigNumber, Contract } from 'ethers'
 import { Operator, StreamrConfig, operatorABI, streamrConfigABI } from '@streamr/network-contracts'
 import { OperatorServiceConfig } from './OperatorPlugin'
+import { EthereumAddress } from 'streamr-client'
+import { Logger, TheGraphClient, toEthereumAddress } from '@streamr/utils'
+import fetch from 'node-fetch'
+
+const logger = new Logger(module)
+
+const ONE_ETHER = BigInt(1e18)
+
+interface UnwithdrawnEarningsData {
+    sumDataWei: bigint
+    fraction: bigint
+    sponsorshipAddresses: EthereumAddress[]
+}
 
 export class MaintainOperatorValueHelper {
     private readonly operator: Operator
+    private readonly theGraphClient: TheGraphClient
     private readonly config: OperatorServiceConfig
 
     constructor(config: OperatorServiceConfig) {
         this.config = config
         this.operator = new Contract(config.operatorContractAddress, operatorABI, this.config.signer) as unknown as Operator
+        this.theGraphClient = new TheGraphClient({
+            serverUrl: config.theGraphUrl,
+            fetch,
+            logger
+        })
     }
 
-    // returns a wei value (1 ETH means 100%)
+    async getRandomOperator(): Promise<EthereumAddress | undefined> {
+        const latestBlock = await this.operator.provider.getBlockNumber()
+        const queryFilter = '' // e.g. (first: 10, orderBy: poolValue, orderDirection: desc)
+        const operators = await this.getOperatorAddresses(latestBlock, queryFilter)
+        // filter out my own operator
+        const operatorAddresses = operators.filter((id) => toEthereumAddress(id) !== toEthereumAddress(this.config.operatorContractAddress))
```

Comment:
both of the comparison operands have already the type of `EthereumAddress`

```suggestion
        const operatorAddresses = operators.filter((id) => id !== this.config.operatorContractAddress)
```